[BUG] Compiler error when attempting to update to ESPHome 2025.11

The following errors prevent compilation when updating from ESPHome 2025.10.4 to ESPHome 2025.11

```
In file included from src/esphome.h:108,
                 from src/main.cpp:3:
src/esphome/components/xiaomi_bslamp2/light/automation.h: In instantiation of 'class esphome::xiaomi::bslamp2::ActivatePresetAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >':
/config/_my/_esphome/packages/p-kz-ble-sleeplight/behavior_default.yaml:128:146:   required from here
  128 |   # been handled above (by holding the power button). Therefore, brightness
      |                                                                                                                                                  ^
src/esphome/components/xiaomi_bslamp2/light/automation.h:86:8: error: 'void esphome::xiaomi::bslamp2::ActivatePresetAction<Ts>::play(Ts ...) [with Ts = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]' marked 'override', but does not override
   86 |   void play(Ts... x) override {
      |        ^~~~
/config/_my/_esphome/packages/p-kz-ble-sleeplight/behavior_default.yaml: In function 'void setup()':
/config/_my/_esphome/packages/p-kz-ble-sleeplight/behavior_default.yaml:128:146: error: invalid new-expression of abstract class type 'esphome::xiaomi::bslamp2::ActivatePresetAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >'
  128 |   # been handled above (by holding the power button). Therefore, brightness
      |                                                                                                                                                  ^
src/esphome/components/xiaomi_bslamp2/light/automation.h:78:32: note:   because the following virtual functions are pure within 'esphome::xiaomi::bslamp2::ActivatePresetAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >':
   78 | template<typename... Ts> class ActivatePresetAction : public Action<Ts...> {
      |                                ^~~~~~~~~~~~~~~~~~~~
In file included from src/esphome/components/binary_sensor/filter.h:3,
                 from src/esphome/components/binary_sensor/binary_sensor.h:5,
                 from src/esphome/core/application.h:30,
                 from src/esphome/components/api/api_frame_helper.h:13,
                 from src/esphome/components/api/api_connection.h:5,
                 from src/esphome.h:3:
src/esphome/core/automation.h:225:16: note:     'void esphome::Action<Ts>::play(const Ts& ...) [with Ts = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]'
  225 |   virtual void play(const Ts &...x) = 0;
      |                ^~~~
```

I think my install log looks slightly different than OP's so I'm pasting mine here just in case:

```
INFO ESPHome 2025.11.0
INFO Reading configuration /config/esphome/bedside-lamp.yaml...
INFO Updating https://github.com/mmakaay/esphome-xiaomi_bslamp2@dev
INFO Updating https://github.com/mmakaay/esphome-xiaomi_bslamp2@dev
INFO Generating C++ source...
INFO Compiling app... Build path: /data/build/bedside-lamp
Processing bedside-lamp (board: esp32doit-devkit-v1; framework: espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/55.03.31-2/platform-espressif32.zip)
--------------------------------------------------------------------------------
INFO Package configuration completed successfully
INFO Package configuration completed successfully
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - contrib-piohome @ 3.4.4 
 - framework-espidf @ 3.50501.0 (5.5.1) 
 - tool-cmake @ 4.0.3 
 - tool-esp-rom-elfs @ 2024.10.11 
 - tool-esptoolpy @ 5.1.0 
 - tool-mklittlefs @ 3.2.0 
 - tool-ninja @ 1.13.1 
 - tool-scons @ 4.40801.0 (4.8.1) 
 - toolchain-xtensa-esp-elf @ 14.2.0+20241119
Reading CMake configuration...
Dependency Graph
|-- noise-c @ 0.1.10
Compiling .pioenvs/bedside-lamp/src/main.cpp.o
Archiving .pioenvs/bedside-lamp/esp-idf/efuse/libefuse.a
Compiling .pioenvs/bedside-lamp/esp-tls/esp-tls-crypto/esp_tls_crypto.c.o
Compiling .pioenvs/bedside-lamp/esp-tls/esp_tls_error_capture.c.o
Compiling .pioenvs/bedside-lamp/esp-tls/esp_tls_platform_port.c.o
Compiling .pioenvs/bedside-lamp/esp-tls/esp_tls_mbedtls.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/adc_oneshot.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/adc_common.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/adc_cali.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/adc_cali_curve_fitting.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/deprecated/esp_adc_cal_common_legacy.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/adc_continuous.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/esp32/adc_dma.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/esp32/adc_cali_line_fitting.c.o
Compiling .pioenvs/bedside-lamp/esp_adc/deprecated/esp32/esp_adc_cal_legacy.c.o
Compiling .pioenvs/bedside-lamp/esp_app_format/esp_app_desc.c.o
Compiling .pioenvs/bedside-lamp/esp_bootloader_format/esp_bootloader_desc.c.o
Compiling .pioenvs/bedside-lamp/esp_coex/esp32/esp_coex_adapter.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp-tls/libesp-tls.a
Archiving .pioenvs/bedside-lamp/esp-idf/esp_app_format/libesp_app_format.a
Archiving .pioenvs/bedside-lamp/esp-idf/esp_bootloader_format/libesp_bootloader_format.a
Archiving .pioenvs/bedside-lamp/esp-idf/esp_adc/libesp_adc.a
Compiling .pioenvs/bedside-lamp/esp_coex/src/coexist_debug_diagram.c.o
Compiling .pioenvs/bedside-lamp/esp_coex/src/coexist_debug.c.o
Compiling .pioenvs/bedside-lamp/esp_common/src/esp_err_to_name.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp_coex/libesp_coex.a
Compiling .pioenvs/bedside-lamp/esp_driver_cam/esp_cam_ctlr.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_cam/dvp_share_ctrl.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_dac/dac_oneshot.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_dac/dac_cosine.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp_driver_cam/libesp_driver_cam.a
Compiling .pioenvs/bedside-lamp/esp_driver_dac/dac_continuous.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_dac/dac_common.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp_common/libesp_common.a
Compiling .pioenvs/bedside-lamp/esp_driver_dac/esp32/dac_dma.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_gpio/src/gpio.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_gpio/src/gpio_glitch_filter_ops.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_gpio/src/rtc_io.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_gptimer/src/gptimer.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_gptimer/src/gptimer_common.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp_driver_dac/libesp_driver_dac.a
Compiling .pioenvs/bedside-lamp/esp_driver_i2c/i2c_master.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_i2c/i2c_common.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_i2c/i2c_slave.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp_driver_gptimer/libesp_driver_gptimer.a
Compiling .pioenvs/bedside-lamp/esp_driver_i2s/i2s_common.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp_driver_gpio/libesp_driver_gpio.a
Compiling .pioenvs/bedside-lamp/esp_driver_i2s/i2s_std.c.o
In file included from src/esphome.h:97,
                 from src/main.cpp:3:
src/esphome/components/xiaomi_bslamp2/output/automation.h: In instantiation of 'class esphome::xiaomi::bslamp2::SetLevelAction<float>':
/data/packages/c1741637/packages/behavior_default.yaml:214:110:   required from here
src/esphome/components/xiaomi_bslamp2/output/automation.h:46:8: error: 'void esphome::xiaomi::bslamp2::SetLevelAction<Ts>::play(Ts ...) [with Ts = {float}]' marked 'override', but does not override
   46 |   void play(Ts... x) override {
      |        ^~~~
/data/packages/c1741637/packages/behavior_default.yaml: In function 'void setup()':
/data/packages/c1741637/packages/behavior_default.yaml:214:110: error: invalid new-expression of abstract class type 'esphome::xiaomi::bslamp2::SetLevelAction<float>'
src/esphome/components/xiaomi_bslamp2/output/automation.h:40:32: note:   because the following virtual functions are pure within 'esphome::xiaomi::bslamp2::SetLevelAction<float>':
   40 | template<typename... Ts> class SetLevelAction : public Action<Ts...> {
      |                                ^~~~~~~~~~~~~~
In file included from src/esphome/components/binary_sensor/filter.h:3,
                 from src/esphome/components/binary_sensor/binary_sensor.h:5,
                 from src/esphome/core/application.h:30,
                 from src/esphome/components/api/api_frame_helper.h:13,
                 from src/esphome/components/api/api_connection.h:5,
                 from src/esphome.h:3:
src/esphome/core/automation.h:225:16: note:     'void esphome::Action<Ts>::play(const Ts& ...) [with Ts = {float}]'
  225 |   virtual void play(const Ts &...x) = 0;
      |                ^~~~
src/esphome/components/xiaomi_bslamp2/output/automation.h: In instantiation of 'class esphome::xiaomi::bslamp2::UpdateLEDsAction<float>':
/data/packages/c1741637/packages/ota_feedback.yaml:23:114:   required from here
   23 |   on_end:
      |                                                                                                                  ^
src/esphome/components/xiaomi_bslamp2/output/automation.h:59:8: error: 'void esphome::xiaomi::bslamp2::UpdateLEDsAction<Ts>::play(Ts ...) [with Ts = {float}]' marked 'override', but does not override
   59 |   void play(Ts... x) override {
      |        ^~~~
/data/packages/c1741637/packages/ota_feedback.yaml:23:114: error: invalid new-expression of abstract class type 'esphome::xiaomi::bslamp2::UpdateLEDsAction<float>'
   23 |   on_end:
      |                                                                                                                  ^
src/esphome/components/xiaomi_bslamp2/output/automation.h:55:32: note:   because the following virtual functions are pure within 'esphome::xiaomi::bslamp2::UpdateLEDsAction<float>':
   55 | template<typename... Ts> class UpdateLEDsAction : public Action<Ts...> {
      |                                ^~~~~~~~~~~~~~~~
src/esphome/core/automation.h:225:16: note:     'void esphome::Action<Ts>::play(const Ts& ...) [with Ts = {float}]'
  225 |   virtual void play(const Ts &...x) = 0;
      |                ^~~~
In file included from src/esphome.h:84:
src/esphome/components/xiaomi_bslamp2/light/automation.h: In instantiation of 'class esphome::xiaomi::bslamp2::DiscoAction<unsigned char>':
/data/packages/c1741637/packages/ota_feedback.yaml:37:87:   required from here
   37 |           red: 100%
      |                                                                                       ^
src/esphome/components/xiaomi_bslamp2/light/automation.h:49:8: error: 'void esphome::xiaomi::bslamp2::DiscoAction<Ts>::play(Ts ...) [with Ts = {unsigned char}]' marked 'override', but does not override
   49 |   void play(Ts... x) override {
      |        ^~~~
/data/packages/c1741637/packages/ota_feedback.yaml:37:87: error: invalid new-expression of abstract class type 'esphome::xiaomi::bslamp2::DiscoAction<unsigned char>'
   37 |           red: 100%
      |                                                                                       ^
src/esphome/components/xiaomi_bslamp2/light/automation.h:34:32: note:   because the following virtual functions are pure within 'esphome::xiaomi::bslamp2::DiscoAction<unsigned char>':
   34 | template<typename... Ts> class DiscoAction : public Action<Ts...> {
      |                                ^~~~~~~~~~~
src/esphome/core/automation.h:225:16: note:     'void esphome::Action<Ts>::play(const Ts& ...) [with Ts = {unsigned char}]'
  225 |   virtual void play(const Ts &...x) = 0;
      |                ^~~~
/data/packages/c1741637/packages/ota_feedback.yaml:47:87: error: invalid new-expression of abstract class type 'esphome::xiaomi::bslamp2::DiscoAction<unsigned char>'
src/esphome/components/xiaomi_bslamp2/light/automation.h: In instantiation of 'class esphome::xiaomi::bslamp2::ActivatePresetAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >':
/data/packages/c1741637/packages/behavior_default.yaml:152:146:   required from here
  152 |   # been handled above (by holding the power button). Therefore, brightness
      |                                                                                                                                                  ^
src/esphome/components/xiaomi_bslamp2/light/automation.h:86:8: error: 'void esphome::xiaomi::bslamp2::ActivatePresetAction<Ts>::play(Ts ...) [with Ts = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]' marked 'override', but does not override
   86 |   void play(Ts... x) override {
      |        ^~~~
/data/packages/c1741637/packages/behavior_default.yaml:152:146: error: invalid new-expression of abstract class type 'esphome::xiaomi::bslamp2::ActivatePresetAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >'
  152 |   # been handled above (by holding the power button). Therefore, brightness
      |                                                                                                                                                  ^
src/esphome/components/xiaomi_bslamp2/light/automation.h:78:32: note:   because the following virtual functions are pure within 'esphome::xiaomi::bslamp2::ActivatePresetAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >':
   78 | template<typename... Ts> class ActivatePresetAction : public Action<Ts...> {
      |                                ^~~~~~~~~~~~~~~~~~~~
src/esphome/core/automation.h:225:16: note:     'void esphome::Action<Ts>::play(const Ts& ...) [with Ts = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]'
  225 |   virtual void play(const Ts &...x) = 0;
      |                ^~~~
Compiling .pioenvs/bedside-lamp/esp_driver_i2s/i2s_pdm.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_i2s/i2s_platform.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_ledc/src/ledc.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_mcpwm/src/mcpwm_cap.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_mcpwm/src/mcpwm_cmpr.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp_driver_i2s/libesp_driver_i2s.a
Compiling .pioenvs/bedside-lamp/esp_driver_mcpwm/src/mcpwm_com.c.o
Archiving .pioenvs/bedside-lamp/esp-idf/esp_driver_i2c/libesp_driver_i2c.a
Compiling .pioenvs/bedside-lamp/esp_driver_mcpwm/src/mcpwm_fault.c.o
Compiling .pioenvs/bedside-lamp/esp_driver_mcpwm/src/mcpwm_gen.c.o
*** [.pioenvs/bedside-lamp/src/main.cpp.o] Error 1
Compiling .pioenvs/bedside-lamp/esp_driver_mcpwm/src/mcpwm_oper.c.o
========================== [FAILED] Took 7.35 seconds ==========================
```

--- a/components/xiaomi_bslamp2/light/automation.h
+++ b/components/xiaomi_bslamp2/light/automation.h
@@ -46,7 +46,7 @@ template<typename... Ts> class DiscoAction : public Action<Ts...> {
   TEMPLATABLE_VALUE(float, color_temperature)
   TEMPLATABLE_VALUE(std::string, effect)
 
-  void play(Ts... x) override {
+  void play(const Ts &...x) override {
     if (this->disco_state_.has_value()) {
       auto p = this->disco_state_.optional_value(x...);
       if (!*p) {
@@ -83,7 +83,7 @@ template<typename... Ts> class ActivatePresetAction : public Action<Ts...> {
   TEMPLATABLE_VALUE(std::string, group);
   TEMPLATABLE_VALUE(std::string, preset);
 
-  void play(Ts... x) override {
+  void play(const Ts &...x) override {
     auto operation = this->operation_.value(x...);
 
     if (operation == "next_group") {

--- a/components/xiaomi_bslamp2/output/automation.h
+++ b/components/xiaomi_bslamp2/output/automation.h
@@ -17,7 +17,7 @@ template<typename... Ts> class SetLEDsAction : public Action<Ts...> {
   TEMPLATABLE_VALUE(int, mode)
   TEMPLATABLE_VALUE(uint16_t, leds)
 
-  void play(Ts... x) override {
+  void play(const Ts &...x) override {
     uint16_t mode = this->mode_.value(x...);
     uint16_t value = this->leds_.value(x...);
     switch (mode) {
@@ -43,7 +43,7 @@ template<typename... Ts> class SetLevelAction : public Action<Ts...> {
 
   TEMPLATABLE_VALUE(float, level)
 
-  void play(Ts... x) override {
+  void play(const Ts &...x) override {
     parent_->set_level(this->level_.value(x...));
     parent_->update_leds();
   }
@@ -56,7 +56,7 @@ template<typename... Ts> class UpdateLEDsAction : public Action<Ts...> {
  public:
   explicit UpdateLEDsAction(XiaomiBslamp2FrontPanelOutput *parent) : parent_(parent) {}
 
-  void play(Ts... x) override {
+  void play(const Ts &...x) override {
     parent_->update_leds();
   }
 

